sunpy/net/dataretriever/sources/swap.py

```diff
@@ -59,12 +58,18 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         returns list of urls corresponding to given TimeRange.
         """
         level = kwargs.get('level', 1)
-        SWAP_STARTDATE = datetime.datetime(2009, 11, 24)
+        if (level == 1 or level == 0):
+            SWAP_STARTDATE = datetime.datetime(2009, 11, 24)
+        else:
+            SWAP_STARTDATE = datetime.datetime(2010, 1, 4)
         if timerange.start < SWAP_STARTDATE:
             raise ValueError('Earliest date for which SWAP data is available is {:%Y-%m-%d}'.format(SWAP_STARTDATE))
-        datatype = {0: 'eng', 1:'bsd'}
-        prefix = 'http://proba2.oma.be/swap/data/{datatype}/'
-        suffix = '%Y/%m/%d/{instrument}_lv{level}_%Y%m%d_%H%M%S.fits'
+        datatype = {0: 'eng', 1:'bsd', 'q':'qlk', 'Q':'qlk'}
+        prefix = 'http://proba2.oma.be/swap/data/{datatype}/%Y/%m/%d/'
+        if (level == 0 or level == 1):
+            suffix = '{instrument}_lv{level}_%Y%m%d_%H%M%S.fits'
+        else:
+            suffix = '%Y_%m_%d__%H_%M_%S__PROBA2_SWAP_SWAP_174.jp2'
```

Comment:
@dpshelio: The error checking is done in ```_can_handle_query``` so If user enters any other letter apart from 0,1 or 'q' ```_can_handle_query``` will throw an error.